src/main/java/com/example/springtweet/springtweet/controller/TwitterController.java

```diff
@@ -0,0 +1,47 @@
+package com.example.springtweet.springtweet.controller;
+
+import com.example.springtweet.springtweet.exceptionHandler.CustomException;
+import com.example.springtweet.springtweet.exceptionHandler.SuccessMessage;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import twitter4j.*;
+
+import java.io.File;
+
+@RestController
+@RequestMapping(value = "/api/1.0/twitter")
+public class TwitterController {
+
+    Twitter twitter = TwitterFactory.getSingleton();
+
+    @GetMapping(value = "/timeline")
+    public ResponseList<Status> getTimeLine() {
+        try {
+            return twitter.getHomeTimeline();
+        } catch (TwitterException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @PostMapping(value = "/tweet")
+    private ResponseEntity postTweet(@RequestBody String tweetMessage) throws SuccessMessage, CustomException {
+
+        try {
+            Twitter twitter = new TwitterFactory().getInstance();
+            StatusUpdate status = new StatusUpdate(tweetMessage);
+            status.setMedia(new File("/Users/gokul.ravichandran/Desktop/sc.png"));
+            Status updatesStatus = twitter.updateStatus(status);
+            String url = null;
+            MediaEntity[] mediaEntities = updatesStatus.getMediaEntities();
+            for (MediaEntity m : mediaEntities) {
+                url = m.getURL();
```

Comment:
There seems to be a much easier way of building the URL of the tweet: https://stackoverflow.com/questions/22045720/twitter4j-how-to-get-twit-url-from-status